solr/core/src/test/org/apache/hadoop/http/HttpServer2.java

```diff
@@ -0,0 +1,1685 @@
+/*
```

Comment:
This file is copied directly from Apache Hadoop to allow integration tests to run under Jetty 9.4.

I couldn't find a way to get the shaded Hadoop dependencies (new in Hadoop 3) to work since they shaded too much (like javax.servlet). This made our integration with hadoop-auth break.

I have no idea if copying HttpServer2 in here will break other things. I tested with JDK 8 and so far this looks good.

Reply:
@uschindler I know you have a lot of experience with JDK 9+ and packaging. Is this even allowed to have a separate package `org.apache.hadoop...` inside the Solr test directory? Will it cause issues with other JDK versions?